src/parser/flags.ts

```diff
@@ -35,13 +35,21 @@ export function boolean<T = boolean>(
   } as BooleanFlag<T>
 }
 
-export const integer = build({
-  parse: async input => {
-    if (!/^-?\d+$/.test(input))
-      throw new Error(`Expected an integer but received: ${input}`)
-    return Number.parseInt(input, 10)
-  },
-})
+export const integer = (opts: { min?: bigint; max?: bigint } & Partial<OptionFlag<number>> = {}): OptionFlag<number | undefined> => {
+  return build<number>({
+    ...opts,
+    parse: async input => {
+      if (!/^-?\d+$/.test(input))
+        throw new Error(`Expected an integer but received: ${input}`)
+      const inputInt = BigInt(input)
+      if (typeof opts.min === 'bigint' && inputInt < opts.min)
+        throw new Error(`Expected an integer greater than or equal to ${opts.min} but received: ${input}`)
+      if (typeof opts.max === 'bigint' && inputInt > opts.max)
+        throw new Error(`Expected an integer less than or equal to ${opts.max} but received: ${input}`)
+      return Number(inputInt)
```

Comment:
Why make these BigInts instead of normal numbers?
I'd recommend against it unless there's a good reason, and if so we should put it in a comment. Also, if they have to be BigInts we shouldn't cast them back to a Number in the return statement because they can lose precision that way.

There's a few pitfalls associated with BigInts, [see this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt).

Reply:
"Pete said to" is really all I got.  I'd prefer my original version with numbers.  Will switch it back now.

They needed to end up as a number to avoid breaking changes to the oclif integer flag, which results in type `number`

Reply:
Sounds good. If need be, we can just create a `BigInt` flag instead of changing this one.